src/stream/src/executor/row_id_gen.rs

```diff
@@ -63,17 +65,38 @@ impl RowIdGenExecutor {
         let mut builder = SerialArrayBuilder::new(len);
 
         for ((datum, op), vis) in column.iter().zip_eq_fast(ops).zip_eq_fast(vis.iter()) {
-            // Only refill row_id for insert operation.
-            match op {
-                Op::Insert => builder.append(Some(self.row_id_generator.next().into())),
-                _ => {
-                    if vis {
-                        builder.append(Some(Serial::try_from(datum.unwrap()).unwrap()))
-                    } else {
-                        builder.append(None)
```

Comment:
We only want to generate row id for `INSERT`s instead of `UPDATE`s and `DELETE`s. To achieve this, we determine it by `Op`. However, with upsert DML `UPDATE`s will also yield `Op::Insert`.

Thus, this PR changed the logic to skipping generating once there's a non-NULL value for row id column, which also seems more reasonable.